File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var ans:Bool = false
    var st = [Character]()
    for c in s {
        if c == "(" { st.append(c) }
        else {
            if !st.isEmpty && st.last! == "(" {
                st.removeLast();
            } else {
                st.append(c)
            }
        }
    }

    return st.isEmpty
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> st = new Stack<>();
        for(char ss: s.toCharArray()) {
            if (ss == ')') {
                if (!st.isEmpty() && st.peek() == '(') st.pop();
                else st.push(ss);
            }
            else st.push(ss);
        }

        return st.isEmpty() ? true : false;
    }
}